crates/router/src/connector/bankofamerica/transformers.rs

```diff
@@ -377,12 +796,46 @@ impl
             &BankOfAmericaConsumerAuthValidateResponse,
         ),
     ) -> Self {
+        let (action_list, action_token_types, authorization_options) = if item
+            .router_data
+            .request
+            .is_merchant_initiated_mandate_payment()
+        {
+            (
+                Some(vec![BankOfAmericaActionsList::TokenCreate]),
+                Some(vec![BankOfAmericaActionsTokenType::PaymentInstrument]),
+                Some(BankOfAmericaAuthorizationOptions {
+                    initiator: Some(BankOfAmericaPaymentInitiator {
+                        initiator_type: Some(BankOfAmericaPaymentInitiatorTypes::Customer),
+                        credential_stored_on_file: Some(true),
+                        stored_credential_used: None,
+                    }),
+                    merchant_intitiated_transaction: None,
+                }),
+            )
+        } else {
+            (None, None, None)
+        };
+
+        // To distinguish between SetupMandate and Authorization flows.
+        let is_capture_automatic = matches!(
+            item.router_data.request.capture_method,
+            Some(enums::CaptureMethod::Automatic) | None
+        );
+        let is_setup_mandate_payment = item.router_data.request.is_setup_mandate_payment();
+
+        let capture = Some(
+            !((is_setup_mandate_payment && is_capture_automatic)
+                || (!is_setup_mandate_payment && !is_capture_automatic)),
+        );
+
```

Comment:
Check this logic and maybe try to keep this logic at connector level on the basis of amount.